### PR TITLE
Feat/analytics service

### PR DESCRIPTION
**Pull request by harshitpruthi94**
## Types of changes

What types of changes does your code introduce to this project?

_Put an `x` in the boxes that apply_

- [ ] Bugfix (non-breaking change which fixes an issue)

- [ ] New feature (non-breaking change which adds functionality)

- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist before merging

_Put an `x` in the boxes that apply_

- [ ] These changes have been thoroughly tested.

- [ ] Changes need to be immediately published on npm.

### CODE REVIEW

File: packages/web-analytics/.eslintignore [hunk 1]
```diff
@@ -0,0 +1,7 @@
+# npm
```
Comment (shubhamgavali1998) [inline packages/web-analytics/.eslintignore:1]: add pnpm-lock

File: packages/web-analytics/package.json [hunk 2]
```diff
@@ -0,0 +1,49 @@
+{
+  "name": "@groww-tech/web-storage",
```
Comment (shubhamgavali1998) [inline packages/web-analytics/package.json:2]: change name

File: packages/web-analytics/package.json [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+  "license": "MIT",
+  "publishConfig": {
+    "access": "public"
+  },
+  "bugs": {
+    "url": "https://github.com/Groww/webster/issues"
+  },
+  "homepage": "https://github.com/Groww/webster/tree/develop/packages/web-analytics#readme",
+  "devDependencies": {
+    "@groww-tech/eslint-config": "^1.0.4",
```
Comment (shubhamgavali1998) [inline packages/web-analytics/package.json:40]: use `workspace:*`

File: packages/web-analytics/src/index.ts [hunk 4]
```diff
@@ -0,0 +1,172 @@
...
+ * @param thirdPartyId This is the unique id which we send which maps it to a specific user
+ * @param phoneNumber This is the phone number of the user
+ *
+ *
+ * @example
+ * ```
+ * identifyLoggedInUser('John Doe', 'johndoe@gmail.com', 'test123', '9876543210')
+ * ```
+ */
+export function identifyLoggedInUser(name: string, emailId: string, thirdPartyId = '', phoneNumber = '') {
```
Comment (shubhamgavali1998) [inline packages/web-analytics/src/index.ts:114]: add a function

File: packages/web-analytics/src/index.ts [hunk 5]
```diff
@@ -0,0 +1,209 @@
...
+  identifyWebengageLoggedInUser(name, emailId, thirdPartyId, phoneNumber);
+}
+
+
+/**
+ *
```
Comment (tusharsehwagg) [inline packages/web-analytics/src/index.ts:131]: params not added.

File: packages/web-analytics/src/index.ts [hunk 6]
```diff
@@ -0,0 +1,209 @@
...
+      countWebengageLoad++;
+      setTimeout(() => updateAttributeInWebengage(attribute, value), 1000);
+    }
+  }
+}
+
+
+/**
+ *
+ * @returns This function checks and returns that webengage is not undefined
```
Comment (tusharsehwagg) [inline packages/web-analytics/src/index.ts:202]: return type is boolean. We can specify what function goes above it.

**Review by tusharsehwagg — CHANGES_REQUESTED** [review]

**Review by tusharsehwagg — APPROVED** [review]

File: packages/ella/package.json [hunk 7]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "@groww-tech/ella",
-  "version": "0.2.4",
+  "version": "0.2.5",
```
Comment (harshitpruthi94) [inline packages/ella/package.json:3]: Have increased this version as there is a dependency on ella

Comment (vikaz-singh) [reply packages/ella/package.json:3]: If dependency on Ella then add Ella as dependency, why increasing the version?

File: README.md [hunk 8]
```diff
@@ -30,6 +30,7 @@ This monorepo contains the following packages:
 - `stylelint-config`: Standard Stylelint config adopted in Groww. Customized as per requirement and preferences of devs in Groww. [![npm version](https://badge.fury.io/js/@groww-tech%2Fstylelint-config.svg)](https://badge.fury.io/js/@groww-tech%2Fstylelint-config)
 - `ui-toolkit`: Standard UI Library adopted in Groww. Customized as per requirement and preferences of devs in Groww. [![npm version](https://badge.fury.io/js/@groww-tech%2Fui-toolkit.svg)](https://badge.fury.io/js/@groww-tech%2Fui-toolkit)
 - `ella`: Utility-belt library for JavaScript that provides general purpose methods used in day to day programming. [![npm version](https://badge.fury.io/js/@groww-tech%2Fella.svg)](https://badge.fury.io/js/@groww-tech%2Fella)
+- `web-analytics`: Web Analytics is a service used that exposes methods to send events to 3rd party analytics tools like Webengage and Gtm. [![npm version](https://badge.fury.io/js/@groww-tech%2Fweb-analytics.svg)](https://badge.fury.io/js/@groww-tech%2Fweb-analytics)
```
[the remark this answers is not in the document]
Comment (sandeep-groww) [reply README.md:33]: Added 
https://github.com/Groww/webster/pull/187

**Pull request merged**